drivers/serial/uart_nrfx_uarte.c

```diff
@@ -632,13 +632,13 @@ static void rx_timeout(struct k_timer *timer)
 	}
 
 	/* Check if there is data that was not sent to user yet */
-	if (data->async->rx_total_byte_cnt
-	    != data->async->rx_total_user_byte_cnt) {
+
+	u32_t len = data->async->rx_total_byte_cnt
```

Comment:
shouldn't that be `s32_t`?
according to the issue report, it is possible that user_byte_cnt can be larger than total:

>This happens when rx_total_user_byte_cnt becomes larger than rx_total_byte_cnt.